### PR TITLE
Testing - Enumerables Methods with Rspec

### PR DESCRIPTION
**Pull request by victor-aldaraca**
This project involves the use of rspec to test our previous implemented methods.
I installed all gems required for this project and tested all methods to pass all test cases by refactoring when needed.

### CODE REVIEW

File: spec/spec_enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+describe Enumerable do
+  let(:friends) { %w[Sharon Leo Leila Brian Arun Jean] }
+
+  describe 'my_each' do
+    it 'returns to_enum if no arguments or block were given' do
+      expect([1, 2, 3].my_each).to be_a Enumerator
+    end
+    it 'returns the array when a block is given' do
+      expect([1, 2, 3].my_each { |ele| }).to eql([1, 2, 3])
+    end
```
Comment (jcy2704) [inline spec/spec_enumerables.rb]: For example here. You are not implementing the function `#my_each` correctly. Why? You have an empty block. Instead, find a way to implement your `each` function. For this one, I will give you one example, but for the others with the same issue, you need to look or make one yourself.

```
it '' do
  arr = []
  [1, 2, 3].my_each { |ele| arr.push(ele) }
  expect(arr).to eql([1, 2, 3])
end
```

Or you can also still use your test, but don't leave the block empty. You can add `p ele` so it is not empty, in the end, it still returns the main object `[1, 2, 3]`.

- [x] Implement correctly

File: spec/spec_enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  end
+
+  describe 'my_each_with_index' do
+    it 'returns to_enum if no arguments or block were given' do
+      expect([1, 2, 3].my_each_with_index).to be_a Enumerator
+    end
+    it 'returns an array with its index if block were given' do
+      expect([1, 2, 3].my_each_with_index { |ele, idx| }).to eql([1, 2, 3])
+    end
```
Comment (jcy2704) [inline spec/spec_enumerables.rb]: - [x] Implement correctly

File: spec/spec_enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    it 'returns an array containing all elements of enum for which the given block returns a true value.' do
+      expect([1, 2, 3, 4, 8].my_select(&:even?)).to eql([2, 4, 8])
+    end
+    it 'If no block is given, an Enumerator is returned instead' do
+      expect([1, 2, 3, 4].my_select).to be_a Enumerator
+    end
+  end
+
+  let(:array_of_strings) { %w[ant bear cat] }
+  let(:eg_range) { (1..15) }
```
Comment (jcy2704) [inline spec/spec_enumerables.rb]: - [x] Put all the variables at the top before all tests.

File: spec/spec_enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  describe '#my_all?' do
+    context 'when object is an array' do
+      it '#my_all?' do
```
Comment (jcy2704) [inline spec/spec_enumerables.rb]: You are already telling that it is from `#my_all?` method you don't need to tell again that it is from the same method. Instead, just make it as you did before with previous tests. `describe '#my_all?' -> it 'when object is array'`

Do it with all the other test cases.

**Review by jcy2704 — APPROVED** [review]
# CHANGES REQUIRED 👀🤔

## Hey @vicmaburrito! You did a great job doing the test cases and all pass! 🙌

### However, we need to make some minor changes to improve some things.

- **At first I was confused about why I wasn't able to run the spec and it was because I notice that the file is not named correctly. Usually one will run the tests with `rspec --format doc` command so it can run all spec files.**

  - [x] Change the file name so it ends with `_spec.rb` for example, `enumerables_spec.rb` this way the rspec reads it as a test file.
  
- **Your test cases all passed, but some tests are implemented incorrectly. (SEE BELOW FOR SUGGESTIONS)**
  
- [x] Add negative test cases.

- [COMMENTS] Try to download the updated linters once in a while, this way it works all perfectly. I entered your project and had a warning about something related to the rubocop.yml. It still works, but just in case change to the updated version once in a while.

<hr>

### Keep the awesome work! 💪 Happy Coding for y'all!! 💻

**If you have any questions you can ask me on my Slack.**

**Best,**

**[💁‍♂️ Steven Jack Chung Yau](https://stevenchung.me)**

[LinkedIn](https://linkedin.com/in/stevenjchung) | [Twitter](https://twitter.com/_yiak) | [GitHub](https://github.com/jcy2704)

**Review by taiwo2 — APPROVED** [review]
Great job
Am really sorry for the inconvenience, I reviewed you code  and i forgot to left a comment. Am really sorry for the inconvenience it its might have caused to you.thanks for your undestanding
Your project is Approved

**Review by sinansevgi — APPROVED** [review]
# Project status: Approved! :heavy_check_mark: 
----

Hello, @vicmaburrito 

Your readme has clear testing instructions :+1: 

You have made all the changes requested by the previous TSE. :confetti_ball: 

You can merge this pull request and continue with your journey :wave: 

--------
[Optional]

- Microverse [linters-config](https://github.com/microverseinc/linters-config/tree/master/ruby/.github/workflows) repository has a GitHub workflow for RSpec (tests.yml). If you add this workflow to your repository, we can see test results on pull-request checks also.

-------
Sinan Sevgi
